spec/javascript/components/organisms/StatusModalWrapper.test.js

```diff
@@ -0,0 +1,127 @@
+    wrapper = getWrapper({ status: 'Lease Signed', isOpen: true })
+    wrapper.update()
+
+    wrapper.find('button.primary').simulate('submit')
+    expect(ON_SUBMIT).toHaveBeenCalled()
+    expect(validateSpy).toHaveBeenCalled()
+  })
+  test('should display the status dropdown with the status provided', () => {
+    wrapper = getWrapper({ status: 'Appealed', isOpen: true })
+    wrapper.update()
+
+    expect(wrapper.find('StatusDropdown').props().status).toBeTruthy()
+  })
+  test('should display correct substatus options if substatuses are available', () => {
+    wrapper = getWrapper({ status: 'Appealed', isOpen: true })
+    wrapper.update()
+
+    expect(wrapper.find('SubstatusDropdown').exists()).toBeTruthy()
+    const expectedAppealedItems = LEASE_UP_SUBSTATUS_OPTIONS.Appealed
+    expect(wrapper.find('SubstatusDropdown').find('Dropdown').props().items).toEqual(expect.arrayContaining(expectedAppealedItems))
+  })
+
+  test('should not display sub status dropdown if substatuses are not available', () => {
+    wrapper = getWrapper({ status: 'Processing', isOpen: true })
+    wrapper.update()
+
+    expect(wrapper.contains('SubstatusDropdown')).toBeFalsy()
+  })
+
+  test('should not require comment if status does not require comment', () => {
+    wrapper = getWrapper({ status: 'Lease Signed', isOpen: true })
+    wrapper.update()
+    wrapper.find('button.primary').simulate('submit')
+
+    expect(wrapper.find('TextAreaField').find('FieldError > span.error').exists()).toBeFalsy()
+    expect(ON_SUBMIT).toHaveBeenCalled()
+  })
+
+  describe('when required fields are missing', () => {
+    test('should not show error state before fields are touched', () => {
+      wrapper = getWrapper({ status: 'Appealed', isOpen: true })
+      wrapper.update()
+
+      expect(wrapper.find('SubstatusDropdown').props().hasError).toBeFalsy()
+      expect(wrapper.find('TextAreaField').find('FieldError > span.error').exists()).toBeFalsy()
+    })
+
+    test('should show error state for substatus and comment on submit if required', () => {
+      wrapper = getWrapper({ status: 'Appealed', isOpen: true })
+      wrapper.update()
+      wrapper.find('button.primary').simulate('submit')
+
+      expect(wrapper.find('SubstatusDropdown').props().hasError).toBe(true)
+      // Expect the comment field to have an error
+      expect(wrapper.find('TextAreaField').find('FieldError > span.error').exists()).toBeTruthy()
+
+      // Expect that we did not try to submit
+      expect(ON_SUBMIT).not.toHaveBeenCalled()
+    })
+
+    test('should stop showing error state and allow submission when required fields are filled out', () => {
+      wrapper = getWrapper({ status: 'Appealed', isOpen: true })
+      wrapper.update()
+      wrapper.find('button.primary').simulate('submit')
+
+      // Verify there are errors initially
+      expect(wrapper.find('SubstatusDropdown').props().hasError).toBe(true)
```

Comment:
I think when it's a component that's already exported it's better to import and pass in the component itself instead of using the string name

```
import SubstatusDropdown from '~/components/molecules/SubstatusDropdown'
...
expect(wrapper.find(SubstatusDropdown).props().hasError).toBe(true) 
```

This comment also goes for `TextAreaField`